ironfish/src/telemetry/telemetry.ts

```diff
@@ -328,4 +329,39 @@ export class Telemetry {
       ],
     })
   }
+
+  submitNewTransactionSeen(transaction: Transaction, seenAt: Date): void {
+    if (this.shouldLogTransaction(transaction)) {
+      this.submit({
+        measurement: 'trasaction_propagation',
+        timestamp: seenAt,
+        tags: [
+          {
+            name: 'hash',
+            value: transaction.hash().toString('hex'),
+          },
+        ],
+        fields: [
+          {
+            name: 'notes',
+            type: 'integer',
+            value: transaction.notesLength(),
+          },
+          {
+            name: 'spends',
+            type: 'integer',
+            value: transaction.spendsLength(),
+          },
+        ],
+      })
+    }
+  }
+
+  /*
+   * We don't want to log all transaction propagation because there are too many
+   * In this way we can only log propagation for a percentage of transactions
+   */
+  private shouldLogTransaction(t: Transaction) {
+    return t.hash().readDoubleBE() % 10000 === 0
```

Comment:
Is this 0.01% of transactions? 3 per block, 180 per hour?

Reply:
With our given throughput (~200 / block) I think this would be about ~80 per hour. Then I'm sure there are some transactions that just never make it to blocks too so maybe slightly more than that

Reply:
That's 15k per 7 days (our influx retention period) so this seems reasonable given our current cardinality of 117k